### PR TITLE
Deprecate set.isIntersecting() and add ! case to deprecation filtering

### PR DESCRIPTION
**Pull request by bmcdonald3**
The same behavior of `set.isIntersecting()` can be achieved through `!set.isDisjoint()`, so the extra method seems extraneous. Additionally, there isn't much of a precedents for this method in other languages, and the ing naming is something that is fairly unique to this method in the Chapel codebase, so we have chosen to deprecate it. This PR also adds the ability to lead a symbol with `!` in the deprecation warning. The need for this was found with the original message that was being used for deprecation, but that message was changed, so this functionality isn't used in this PR, but it was still included to not lose that additional feature.

- [x] paratest

Closes: https://github.com/chapel-lang/chapel/issues/18644

### CODE REVIEW

File: test/deprecated-keyword/messageFiltering.chpl [hunk 1]
```diff
@@ -171,6 +171,11 @@ var x1102 = 1102;
 deprecated "Lorem ipsum :proc:`!abc` dolor sit amet"
```
Comment (lydia-duncan) [inline test/deprecated-keyword/messageFiltering.chpl:171]: It looks like this case already existed - I think that means your cases should be added before line 167 and this line should be removed

**Review by lydia-duncan — APPROVED** [review]
Just a minor alteration to how you've adjusted the test file.

PR message adjustment - I'd change the title to "... and add ! case to deprecation filtering" and clarify the last line in the body so that it's more clear the impact of that change.  Maybe something about how the generated deprecation warning when compiling the code had this additional tag prefix and how that was only meaningful in the context of the chpldoc warning that gets generated for the deprecated function

**Comment by lydia-duncan** [comment]
Also, does that generate a good link for you?  I wasn't able to follow it to get to isDisjoint.  Did you maybe intend that as a code block rather than a link?

**Comment by bmcdonald3** [comment]
Hmm, I am not sure what I am supposed to be looking for with respect to the link/code block. Is there documentation anywhere that explains that type of thing? I was just doing pattern matching against other PRs and don't have a real understanding of how I should expect things to behave at the moment.

**Comment by lydia-duncan** [comment]
The inline markup section of the chpldoc documentation at https://chapel-lang.org/docs/latest/tools/chpldoc/chpldoc.html#inline-markup-2 mentions what ! results in and is the closest we come to describing those sorts of things.  I think we also rely on Sphinx's own documentation of this stuff.

**Comment by bmcdonald3** [comment]
After thinking on this for a bit and understanding the docs a little better, I think I would be happier with a message like this: `"Set isIntersecting() method is deprecated; use the negation of :proc:`isDisjoint` instead"`, so we can still keep the link. It is a bit strange that it says `negation` instead of `!isDisjoint`, but I still think that is more clear due to the formatting oddities and linking in the docs. Does that sound alright with you?

Also, assuming we stick with the new message, should I still include the `!` addition to the AST code? It still seems like a value add, but not very related to this PR anymore then.

**Comment by lydia-duncan** [comment]
That all sounds good to me. I'm okay with you splitting the ! addition to the AST code into a separate PR or leaving it here and just noting that the main change evolved away from using it but that it is still valuable so didn't warrant being dropped on the floor (and would probably lean towards the latter for simplicity)

**Comment by bmcdonald3** [comment]
Sounds good, thank you!

**Comment by cassella** [comment]
Can you not write something like 

`!`:proc:\`isDisjoint\

Or otherwise put the `!` outside the link (but still leave it formatted as code)?  If the message were saying 

`foo(x) is deprecated; use 5+3*bar(x) instead`

I wouldn't expect anything but `bar(x)` to be involved in the link.

**Comment by bmcdonald3** [comment]
@cassella Hmm, that is a good idea, I don't know why I hadn't thought of that sooner... Thanks!

File: modules/standard/Set.chpl [hunk 2]
```diff
@@ -346,7 +346,7 @@ module Set {
       :return: Whether or not this set and `other` intersect.
       :rtype: `bool`
     */
-    deprecated "Set isIntersecting() method is deprecated; use the negation of :proc:`isDisjoint` instead"
+    deprecated "Set isIntersecting() method is deprecated; use the negation of !:proc:`isDisjoint` instead"
```
Comment (cassella) [inline modules/standard/Set.chpl:349]: If the leading `!` works out, it doesn't need to say "the negation of" any more.

**Pull request merged**